hnn_core/parallel_backends.py

```diff
@@ -644,9 +677,25 @@ def simulate(self, net, n_trials, postproc=True):
 
         env = _get_mpi_env()
 
-        sim_data = run_subprocess(command=self.mpi_cmd, obj=net, timeout=4,
-                                  env=env, cwd=os.getcwd(),
-                                  universal_newlines=True)
+        self.proc, sim_data = run_subprocess(
+            command=self.mpi_cmd, obj=net, timeout=30,
+            proc_queue=self.proc_queue, env=env, cwd=os.getcwd(),
+            universal_newlines=True)
 
         dpls = _gather_trial_data(sim_data, net, n_trials, postproc)
         return dpls
+
+    def terminate(self):
+        proc = None
+        try:
+            proc = self.proc_queue.get(timeout=1)
+        except Empty:
+            warn("No currently running process to terminate")
+
+        if proc is not None:
+            proc.terminate()
+            try:
+                proc.wait(5)  # wait maximum of 1s
```

Comment:
5s ?

Reply:
Thanks, yep 5s. Made this change in #297. WIll rebase here.